crates/sui-framework/tests/sui_system_tests.move

```diff
@@ -69,6 +71,105 @@ module sui::sui_system_tests {
+        report_helper(delegatee_address, @0x2, true, scenario);
+        assert!(vector::is_empty(&get_reporters_of(@0x2, scenario)), 0);
+
+        test_scenario::next_tx(scenario, delegatee_address);
+        let cap = test_scenario::take_from_sender<UnverifiedValidatorOperationCap>(scenario);
+        let new_delegatee_address = @0xcafe;
+        transfer::transfer(cap, new_delegatee_address);
+
+        // New delegatee could report validators on behalf of @0x1.
+        report_helper(new_delegatee_address, @0x2, false, scenario);
+        assert!(get_reporters_of(@0x2, scenario) == vector[@0x1], 0);
+
+        // New delegatee could also set reference gas price on behalf of @0x1.
+        set_gas_price_helper(new_delegatee_address, 666, scenario);
+
+        test_scenario::next_tx(scenario, new_delegatee_address);
+        let system_state = test_scenario::take_shared<SuiSystemState>(scenario);
+        let validator = sui_system::active_validator_by_address(&system_state, @0x1);
+        assert!(validator::next_epoch_gas_price(validator) == 666, 0);
+        test_scenario::return_shared(system_state);
+
+        test_scenario::end(scenario_val);
+    }
+
+    #[test]
+    #[expected_failure(abort_code = sui::validator_set::EInvalidCap)]
+    fun test_report_validator_by_delegatee_revoked() {
+        let scenario_val = test_scenario::begin(@0x0);
+        let scenario = &mut scenario_val;
+        set_up_sui_system_state(vector[@0x1, @0x2], scenario);
+
+        // @0x1 transfers the cap object to delegatee.
+        let delegatee_address = @0xbeef;
+        test_scenario::next_tx(scenario, @0x1);
+        let cap = test_scenario::take_from_sender<UnverifiedValidatorOperationCap>(scenario);
+        transfer::transfer(cap, delegatee_address);
+
+        report_helper(delegatee_address, @0x2, false, scenario);
+        assert!(get_reporters_of(@0x2, scenario) == vector[@0x1], 0);
+
+        // @0x1 revokes delegatee's permssion by creating a new
+        // operation cap object.
+        create_new_operation_cap(@0x1, scenario);
+
+        // delegatee no longer has permission to report validators, here it aborts.
+        report_helper(delegatee_address, @0x2, true, scenario);
+
+        test_scenario::end(scenario_val);
+    }
+
+    #[test]
+    #[expected_failure(abort_code = sui::validator_set::EInvalidCap)]
+    fun test_set_reference_gas_price_by_delegatee_revoked() {
+        let scenario_val = test_scenario::begin(@0x0);
+        let scenario = &mut scenario_val;
+        set_up_sui_system_state(vector[@0x1, @0x2], scenario);
+
+        // @0x1 transfers the cap object to delegatee.
+        let delegatee_address = @0xbeef;
+        test_scenario::next_tx(scenario, @0x1);
+        let cap = test_scenario::take_from_sender<UnverifiedValidatorOperationCap>(scenario);
+        transfer::transfer(cap, delegatee_address);
+
+        // With the cap object in hand, delegatee could report validators on behalf of @0x1.
```

Comment:
nit: I think there's some copypasta here -- this comment belongs on the test above.

Reply:
yes I have it on purpose, because these two tests are independent. When this test fails, this comment would give the investigator enough signals of what is expected here.